sql/core/src/test/resources/sql-tests/results/literals.sql.out

```diff
@@ -211,7 +211,7 @@ select 1E309, -1E309
 -- !query 21
 select 0.3, -0.8, .5, -.18, 0.1111, .1111
 -- !query 21 schema
-struct<0.3:decimal(1,1),-0.8:decimal(1,1),0.5:decimal(1,1),-0.18:decimal(2,2),0.1111:decimal(4,4),0.1111:decimal(4,4)>
```

Comment:
Actually, I think this is an existed bug in Spark: for a number less than `1`, its precision&scale are different between `Decimal` and `DecimalType` if it's created from literal(because precision&scale are defined separately). This PR fixes it by:

`private[sql] def fromDecimal(d: Decimal): DecimalType = DecimalType(d.precision, d.scale)`

Reply:
Is this a bug? For example;
```
hive> create table testrel as select 0.3;
hive> describe testrel;
OK
_c0                 	decimal(1,1)   
```
Is it difficult to keep the current behaviour?
cc: @gatorsmile @cloud-fan

Reply:
That's true that there's other DBMS would ignore the leftmost zero, which could bring larger scale for values less than `1`.  I don't know that Spark is also intentionally to follow this. But AFAIK, for number like `0.3`, in Spark, it will have `(precision, scale)` as (2, 1) in `Decimal`, but (1, 1) in `DecimalType`.

Maybe, we shall add this as a new feature in following PR.

Reply:
The precision/scale in decimal type should be the one we expect. Shall we update the underlying `Decimal` and correct the precision?

Reply:
@maropu @cloud-fan I've opened a separate PR #27217 to address this issue. PTAL.